### PR TITLE
First version of submodule example

### PR DESCRIPTION
**Pull request by bethau**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.2)
```
Comment (jankrassnigg) [inline CMakeLists.txt]: I think this should be at least 3.12 better 3.14.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.2)
+
+# include the ez submodule unitilty cmake functions
+include("ezEngine/Code/BuildSystem/CMake/ezUtilsSubmodule.cmake")
+
+ez_detect_languages()
```
Comment (jankrassnigg) [inline CMakeLists.txt:6]: Why is this needed ?

**Review by jankrassnigg — APPROVED** [review]
approved with suggestions

**Review by ChristopherManthei — APPROVED** [review]
